mobility_games/games/musical_cane.py

```diff
@@ -42,39 +50,89 @@ def process_pose(self, msg):    #   Updates position of Tango
                                         msg.pose.orientation.w])
         self.yaw = angles[2]
 
+    def process_cane_tip(self, msg):
+        #   Y is the cane's horizontal position, where y = 0 is the midline of the cane user.
+        self.cane_x = msg.pose.position.x
+        self.cane_y = msg.pose.position.y
+        self.cane_z = msg.pose.position.z
+        angles = euler_from_quaternion([msg.pose.orientation.x,
+                                        msg.pose.orientation.y,
+                                        msg.pose.orientation.z,
+                                        msg.pose.orientation.w])
+        self.yaw = angles[2]
+
     def run(self):
         lps = 3/self.quarter_time    #   loops per second!
         r = rospy.Rate(lps)
         beat = 1
         last_chord = rospy.Time.now()
+        print("Searching for Tango...")
         while not rospy.is_shutdown():
             print self.x
-            if self.x and not self.start:
+            print self.cane_y
+            if self.x and self.cane_x and not self.start:
                 #   Once position signals start to be received,
                 #   note initial position.
                 self.x_init = self.x
                 self.y_init = self.y
                 self.start = True
+                self.cane_y_prev = self.cane_y
                 print("Connection established.")
-            if self.x and self.start and rospy.Time.now() - last_chord >= rospy.Duration(self.quarter_time):
-                dist = math.sqrt((self.x - self.x_init)**2 + (self.y - self.y_init)**2)
+            if self.x and self.cane_x and self.start:
+                if self.cane_y * self.cane_y_prev <= 0:
+                    system("aplay " + os.path.join(self.sound_folder, "beep.wav"))
+                    print("beep")
+                self.cane_y_prev = self.cane_y
+            r.sleep()
+
+    def run_music(self):
```

Comment:
Can you describe in the function doc what the difference between `run_music` and `run`?